bundle/edu.gemini.p2checker/src/main/scala/edu/gemini/p2checker/target/NonSiderealTargetRules.scala

```diff
@@ -0,0 +1,86 @@
+
+import edu.gemini.p2checker.api.{P2Problems, IP2Problems, ObservationElements, IRule}
+import edu.gemini.shared.util.immutable.ScalaConverters._
+import edu.gemini.spModel.core._
+import scalaz._, Scalaz._
+
+final class NonSiderealTargetRules extends IRule {
+  import NonSiderealTargetRules._
+
+  implicit class SemesterOps(s: Semester) {
+    def contains(time: Long, site: Site): Boolean =
+      time >= s.getStartDate(site).getTime &&
+      time <=   s.getEndDate(site).getTime
+  }
+
+  override def check(es: ObservationElements): IP2Problems = {
+    val p2p = new P2Problems
+    p2p.append(checkNoHorizonsDesignation(es))
+    p2p.append(checkNoSchedulingBlock(es))
+    p2p.append(checkSchedulingBlockOutsideSemester(es))
+    p2p.append(checkNoEphemerisForSchedulingBlock(es))
+    p2p
+  }
+
+  def checkNoHorizonsDesignation(es: ObservationElements): IP2Problems =
+    new P2Problems <| { p2p =>
+      for {
+        ocn <- es.getTargetObsComponentNode.asScalaOpt.toList
+        toc <- es.getTargetObsComp.asScalaOpt.toList
+        tar <- toc.getTargetEnvironment.getTargets.asScalaList
+        nst <- tar.getNonSiderealTarget.toList
+               if nst.horizonsDesignation.isEmpty
+      } p2p.addWarning(ERR_NO_HORIZONS_DESIGNATION,
+          s"Target ${nst.name} has no HORIZONS identifier; high-resolution ephemeris cannot be updated automatically.",
+          ocn)
+    }
+
+  def checkNoSchedulingBlock(es: ObservationElements): IP2Problems =
+    new P2Problems <| { p2p =>
+      for {
+        toc <- es.getTargetObsComp.asScalaOpt
+               if toc.getTargetEnvironment.getTargets.asScalaList.exists(_.isNonSidereal)
+      } p2p.addWarning(ERR_NO_SCHEDULING_BLOCK,
+          s"Observation ${es.getObservationNode.getObservationID} has nonsidereal targets but no scheduling block.",
+          es.getObservationNode)
+    }
+
+  def checkSchedulingBlockOutsideSemester(es: ObservationElements): IP2Problems =
+    new P2Problems <| { p2p =>
+      for {
+        toc  <- es.getTargetObsComp.asScalaOpt
+                if toc.getTargetEnvironment.getTargets.asScalaList.exists(_.isNonSidereal)
+        sb   <- es.getObservation.getSchedulingBlock.asScalaOpt
+        site <- Option(es.getObservationNode.getProgramID.site)
+        sem  <- es.getObservationNode.getProgramID.semester
+                if !sem.contains(sb.start, site)
+      } p2p.addWarning(ERR_SCHEDULING_BLOCK_SEM,
+          s"Scheduling block for ${es.getObservationNode.getObservationID} falls outside of ${sem}.",
+          es.getObservationNode)
+    }
+
+  def checkNoEphemerisForSchedulingBlock(es: ObservationElements): IP2Problems =
+    new P2Problems <| { p2p =>
+      for {
+        ocn <- es.getTargetObsComponentNode.asScalaOpt.toList
+        toc <- es.getTargetObsComp.asScalaOpt.toList
+        tar <- toc.getTargetEnvironment.getTargets.asScalaList
+        nst <- tar.getNonSiderealTarget.toList if nst.horizonsDesignation.isEmpty
```

Comment:
Couldn't a target have a horizons designation and yet not have ephemeris including the scheduling block start time?  That is, maybe we don't need the `if nst.horizonsDesignation.isEmpty` filter there or maybe we should but the condition should be reversed?